### DISCUSSION TITLE
Generate type for Layout export component props in the root route

### DISCUSSION THREAD

**Discussion by silvenon** [remix-run/react-router#12500] [Proposals]
This would be a small addition to type generation, but the root route also supports the [`Layout` export](https://reactrouter.com/explanation/special-files#layout-export), it would be nice if generates types for the root route included `LayoutComponentProps` or whatever you'd like to name it.

**Comment by AlemTuzlak** [comment]
Or just have the docs at least say that the props are just `{ children: React.ReactNode }` evem?

**Reply by sergiodxa** [reply]
This could also be a interface RR exports
```ts
import { LayoutProps } from "react-router";
export function Layout(props: LayoutProps) {
  // use props
}
```
That way you don't have to type it yourself, risking a future change may introduce a change to the layout props, but also it's not tied to a route.

**Reply by silvenon** [reply]
True that. On the other hand tying it to the route would make it clear that it can't be used elsewhere because it wouldn't be generated for other routes even if you export it. It's an opportunity for a small safety net, perhaps unnecessary.

**Reply by sergiodxa** [reply]
I think it would be nice if Layout component worked on any routes actually, not just in root, so you can have a shared layout between a route UI and error boundary.

**Reply by silvenon** [reply]
Good point!
